packages/online-editor/src/accelerators/AcceleratorsContext.tsx

```diff
@@ -0,0 +1,414 @@
+          bpmnDestinationFolder: validateAcceleratorDestinationFolderPaths(accelerator.bpmnDestinationFolder),
+          otherFilesDestinationFolder: validateAcceleratorDestinationFolderPaths(
+            accelerator.otherFilesDestinationFolder
+          ),
+        };
+
+        if (
+          destinationPathValidation.dmnDestinationFolder ||
+          destinationPathValidation.bpmnDestinationFolder ||
+          destinationPathValidation.otherFilesDestinationFolder
+        ) {
+          throw destinationPathValidation;
+        }
+
+        const workspaceFiles = await workspaces.getFiles({ workspaceId });
+
+        // Create new temporary branch with current files, but stay on main
+        await workspaces.branch({ workspaceId, name: BACKUP_BRANCH_NAME, checkout: false });
+
+        // Commit moved files to moved files branch
+        await workspaces.commit({
+          workspaceId,
+          commitMessage: "Backup files",
+          targetBranch: BACKUP_BRANCH_NAME,
+        });
+
+        // Create new temporary branch for moved files, but stay on main
+        await workspaces.branch({ workspaceId, name: MOVED_FILES_BRANCH_NAME, checkout: false });
+
+        // Checkout to moved files branch
+        await workspaces.checkout({
+          workspaceId,
+          ref: MOVED_FILES_BRANCH_NAME,
+          remote: GIT_ORIGIN_REMOTE_NAME,
+          force: true,
+        });
+
+        // Move files
+        let currentFileAfterAccelerator: WorkspaceFile | undefined;
+        movedFiles = await Promise.all(
+          workspaceFiles.map(async (file) => {
+            let fileNewDestination: string;
+
+            if (isDecision(file.relativePath) || isScorecard(file.relativePath)) {
+              fileNewDestination = accelerator.dmnDestinationFolder;
+            } else if (isBusinessProcessModel(file.relativePath)) {
+              fileNewDestination = accelerator.bpmnDestinationFolder;
+            } else {
+              fileNewDestination = accelerator.otherFilesDestinationFolder;
+            }
+
+            const movedFile = await workspaces.moveFile({
+              file,
+              newDirPath: join(fileNewDestination, dirname(file.relativePath)),
+            });
+
+            await workspaces.stageFile({ workspaceId, relativePath: movedFile.relativePath });
+
+            if (file.relativePath === currentFile.relativePath) {
+              currentFileAfterAccelerator = movedFile;
+            }
+            return movedFile;
+          })
+        );
+
+        const movedFilesPaths = movedFiles.map((file) => file.relativePath);
+
+        if (!currentFileAfterAccelerator) {
+          throw new Error("Failed to find current file after moving.");
+        }
+
+        // Commit moved files to moved files branch
+        await workspaces.commit({
+          workspaceId,
+          commitMessage: "Moving files to apply Accelerator.",
```

Comment:
```suggestion
          commitMessage: "KIE Sandbox: Moving files to apply Accelerator ${acceleratorName}.",
```

For illustrative purposes only, this code is not correct ^